### PR TITLE
Sync sdk feature branch

### PR DESCRIPTION
**Pull request by wtrocki**
# Sync Feature branch

Sync SDK will using Apollo Link extensible chain to build set of processors for various SDK operations.
This processors will be automatically configured to be build and give users access to sync client object.

screen shot 2018-11-06 at 1 07 42 pm

## Note

Let's use this branch as base for PR's

## Verification

https://github.com/aerogear/offline-conflict-strategies/pull/34

### CODE REVIEW

File: packages/sync/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/aerogear/aerogear-js-sdk.git"
+  },
+  "keywords": [
+    "aerogear",
+    "aerogear services",
+    "graphql"
+  ],
+  "author": "Aero Gear",
```
Comment (danielpassos) [inline packages/sync/package.json]: AeroGear (without space)

File: packages/sync/src/DataSyncClientOptions.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export interface DataSyncClientOptions {
+  /**
+   * Describes name of the field used as ID
+   */
+  defaultObjectIdField?: string;
+
+  /**
+   * Storage solution
+   */
+  storage?: PersistentStore<PersistedData>;
```
Comment (danielpassos) [inline packages/sync/src/DataSyncClientOptions.ts]: Love it!

Comment (wtrocki) [reply packages/sync/src/DataSyncClientOptions.ts]: If you like this like this PR upstream: https://github.com/helfer/apollo-link-queue
Hopefully we could get this merged and get official support for persistence.

File: packages/sync/src/DataSyncClientOptions.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Contains all options required to initialize SDK
+ * @see DefaultOptions for defaults
+ */
+export interface DataSyncClientOptions {
+  /**
+   * Describes name of the field used as ID
+   */
+  defaultObjectIdField?: string;
```
Comment (danielpassos) [inline packages/sync/src/DataSyncClientOptions.ts]: I think we should not use `default` in this field.

File: packages/sync/src/OptionalUiEngine.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+/**
+ * Engine for handling optional UI
+ */
+export class OptionalUIEngine {
```
Comment (danielpassos) [inline packages/sync/src/OptionalUiEngine.ts]: Not clear what is this for.

**Pull request merged**